File: 100joon/Sliver/2606.py
import sys

com = int(input())
n = int(input())
edge = {}
for i in range(1, com + 1):
    edge[i] = []

for _ in range(n):
    a, b = map(int, sys.stdin.readline().split())
    edge[a].append(b)
    edge[b].append(a)

stack = []
visited = {}
cnt = 0

stack.append(1)
while stack:
    current = stack.pop()
    visited[current] = 1
    cnt += 1
    neighbor = edge[current]
    if not neighbor:
        break
    for i in neighbor:
        if not i in visited and not i in stack:
            stack.append(i)

print(cnt - 1)

File: 100joon/Sliver/2606.js
const input = require("fs").readFileSync(0, "utf-8").trim().split("\n");

const n = Number(input[0]);
const m = Number(input[1]);

const edges = Array.from({ length: n + 1 }, () => []);
for (let i = 0; i < m; i++) {
  const [a, b] = input[i + 2].split(" ").map(Number);
  edges[a].push(b);
  edges[b].push(a);
}

let ans = 0;
const stack = [];
const visited = Array(n + 1).fill(false);
visited[1] = true;
stack.push(1);
while (stack.length) {
  x = stack.pop();

  for (let i of edges[x]) {
    if (!visited[i]) {
      visited[i] = true;
      ans += 1;
      stack.push(i);
    }
  }
}

console.log(ans);
